source/common/init/manager_impl.cc

```diff
@@ -16,16 +16,26 @@ void ManagerImpl::add(const Target& target) {
   TargetHandlePtr target_handle(target.createHandle(name_));
   switch (state_) {
   case State::Uninitialized:
-    // If the manager isn't initialized yet, save the target handle to be initialized later.
-    ENVOY_LOG(debug, "added {} to {}", target.name(), name_);
-    target_handles_.push_back(std::move(target_handle));
+    if (target_handle == nullptr) {
```

Comment:
I'd like to handle this a bit differently: I don't think `createHandle` should return `nullptr`, it should return always return a valid handle. That keeps the contract between the `Target` interface and the `ManagerImpl` easiest to reason about, and keeps logic here as simple as possible.

(Imagine, for example, you had decided instead to create a new implementation of `Target` rather than modifying the existing implementation. In that case, you'd probably try to avoid changing `ManagerImpl`).

My suggestion instead is: if the handle points to a target that's already ready, that's fine. When we call `initialize` later, the target should do the right thing (specifically, if it's already ready, it should immediately call back).

Reply:
Make sense! It's a burden to allow returning a nullptr. Will update as per your suggestion.